src/load/azext_load/tests/latest/test_load_test_advanced_urls.py

```diff
@@ -0,0 +1,352 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azext_load.tests.latest.constants import LoadTestConstants
+from azext_load.tests.latest.preparers import LoadTestResourcePreparer
+from azure.cli.testsdk import (
+    JMESPathCheck,
+    ResourceGroupPreparer,
+    ScenarioTest,
+    live_only,
+)
+from knack.log import get_logger
+import urllib.parse
+
+logger = get_logger(__name__)
+
+rg_params = {
+    "name_prefix": "clitest-advnurlload-",
+    "location": "eastus",
+    "key": "resource_group",
+    "parameter_name": "rg",
+    "random_name_length": 30,
+}
+load_params = {
+    "name_prefix": "clitest-advnurlload-",
+    "location": "eastus",
+    "key": "load_test_resource",
+    "parameter_name": "load",
+    "resource_group_key": "resource_group",
+    "random_name_length": 30,
+}
+
+class LoadTestScenarioAdvancedUrl(ScenarioTest):
+    def __init__(self, *args, **kwargs):
+        super(LoadTestScenarioAdvancedUrl, self).__init__(*args, **kwargs)
+        self.kwargs.update({"subscription_id": self.get_subscription_id()})
+    
+    @live_only()
+    @ResourceGroupPreparer(**rg_params)
+    @LoadTestResourcePreparer(**load_params)
+    def test_load_test_advancedurl(self):
+        self.kwargs.update(
+            {
+                "test_id": LoadTestConstants.LOAD_TEST_ADVANCED_URL,
+            }
+        )
+        checks = [
+            JMESPathCheck("testId", LoadTestConstants.LOAD_TEST_ADVANCED_URL),
+            JMESPathCheck("kind", LoadTestConstants.ADVANCED_URL_TEST_TYPE),
+        ]
+        self.cmd(
+            "az load test create "
+            "--test-id {test_id} "
+            "--load-test-resource {load_test_resource} "
+            "--resource-group {resource_group} ",
+            checks=checks,
+        )
+        
+        self.kwargs.update(
```

Comment:
Can we add comments about the specific test scenarios we are testing.. I have to parse it from code.

Reply:
Also is it possible to move scenarios in separate methods. .and call it from the main test method

Reply:
updated